services/autoscaling/src/simcore_service_autoscaling/core/settings.py

```diff
@@ -14,21 +15,57 @@
 from .._meta import API_VERSION, API_VTAG, APP_NAME
 
 
-class AwsSettings(BaseCustomSettings):
-    AWS_KEY_NAME: str
-    AWS_DNS: str
+class EC2Settings(BaseCustomSettings):
+    EC2_ACCESS_KEY_ID: str
+    EC2_ENDPOINT: Optional[str] = Field(
+        default=None, description="do not define if using standard AWS"
+    )
+    EC2_REGION_NAME: str = "us-east-1"
+    EC2_SECRET_ACCESS_KEY: str
+
+
+class EC2InstancesSettings(BaseCustomSettings):
+    EC2_INSTANCES_ALLOWED_TYPES: tuple[str, ...] = Field(
+        ...,
+        description="Defines which EC2 instances are considered as candidates for new docker nodes",
+    )
+    EC2_INSTANCES_AMI_ID: str = Field(
+        ..., description="Defines the AMI ID used to initialize a new docker node"
+    )
+    EC2_INSTANCES_MAX_INSTANCES: int = Field(
+        10,
+        description="Defines the maximum number of instances the autoscaling app may create",
+    )
+    EC2_INSTANCES_SECURITY_GROUP_IDS: list[str] = Field(
+        ..., description="TO BE DEFINED"
+    )
+    EC2_INSTANCES_SUBNET_ID: str = Field(..., description="TO BE DEFINED")
+    EC2_INSTANCES_KEY_NAME: str = Field(
+        ...,
+        description="SSH key filename (without ext) to access the docker swarm manager",
+    )
+
+
+class NodesMonitoringSettings(BaseCustomSettings):
+    NODES_MONITORING_NODE_LABELS: list[str] = Field(
+        default_factory=list,
+        description="autoscaling will only monitor nodes with the given labels (if empty all nodes will be monitored)",
+    )
 
-    AWS_ACCESS_KEY_ID: str
-    AWS_SECRET_ACCESS_KEY: str
-    AWS_REGION_NAME: str = "us-east-1"  # see https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/Concepts.RegionsAndAvailabilityZones.html
+    NODES_MONITORING_SERVICE_LABELS: list[str] = Field(
+        default_factory=list,
+        description="autoscaling will only monitor services with the given labels (if empty all services will be monitored)",
```

Comment:
yes this concerns only docker services, since that is what gets started first and has all the reservations set right?
i.e. the dynamic-sidecar is run as a docker service and has reservations set to be the total amount necessary for the user services. Also, unless going through portainer or maybe the agent I cannot know what all the containers are using on each node.

Reply:
Correct, the dy-sidecar service has all the required reservations.

Reply:
yes but we do not need this since the services are accessible and have all the necessary information.